get_subscription.py

```diff
@@ -87,12 +87,14 @@ def mount_get_root(mount):
 args = parser.parse_args()
 
 try:
-    appids = [app_to_sub[s] for s in args.appids]
+    subids = [app_to_sub[a] for a in args.appids if re.match('^\w+$', a)]
 except KeyError as e:
     logger.critical("couldn't find subid for app: %s", e.args)
     sys.exit()
 
-fetchers = [Fetcher(args.store_dir, appid, face=face).start() for appid in appids]
+assert subids.__len__()
+
+fetchers = [Fetcher(args.store_dir, subid, face=face).start() for subid in subids]
 
 batch = Batch(fetchers, "Subscriptions")
```

Comment:
As mentioned before, there's a race here between `Batch` and `fetchers` that removing `auto=True` would fix.